services/user.js

```diff
@@ -40,13 +40,43 @@ class User {
     async grantToken(params, context){
         isLoggedInOrThrow(context);
         isValidUserStatus(context?.userInfo?.userStatus);
-        let accessToken = createToken(context?.userInfo, config.token_secret, config.tokenTimeout)
+        let accessToken = createToken(context?.userInfo, config.token_secret, config.tokenTimeout);
+        const aUser = await this.linkTokentoUser(context, accessToken);
         return {
             tokens: [accessToken],
             message: "This token can only be viewed once and will be lost if it is not saved by the user"
         }
     }
 
+    async linkTokentoUser(context, accessToken){
+        const sessionCurrentTime = getCurrentTime();
+        const user = await this.userCollection.find(context.userInfo._id);
+        const updateUser ={
+            _id: context.userInfo._id,
+            tokens: [accessToken],
+            updateAt: sessionCurrentTime
+        }
+        const updateResult = await this.userCollection.update(updateUser);
+
+        if (!updateResult?.matchedCount === 1) {
+            throw new Error(ERROR.UPDATE_FAILED);
+        }
+
+        context.userInfo = {
+            ...context.userInfo,
+            ...updateUser,
+            updateAt: sessionCurrentTime
+        }
+        const result = {
```

Comment:
every change below this point can be removed assuming you agree that a return value is not needed